### ISSUE TITLE
Auto-upload of files over 1MB removes timestamp/file-creation-stamp (WLAN; SD-Card)

### ISSUE THREAD

**Issue by EngelPika32** [nextcloud/android#8646]
### Steps to reproduce
1. Create a folder [p][sd]
2. Add the folder to auto-upload [pA] – linking it to a new/empty folder
 -- WLAN only; Upload existing files as well (though, there are no as of now); use Subfolders; Keep original files (can't change that option as the folder is read-only – intended); Ask when duplicated/conflict
3. Place some files like images in it [p][sd]; Filesize: Vary between some KB and some MB.
4. Place some folders with files in the [p][sd]
4. Wait for the auto-upload
5. Check the folder, do they exist? [s] {more in 'Actual behavior'}
6. Compare the timestamps of your files [s] [p] [pA] {more in 'Actual behavior'}

Legend / where was the action taken:
[p] = phone
[pA] = phone, but action taken within the nextcloud app
[s] = server (acced through browser)
[sd] = sd-card
[in] = intern device storage

### Expected behavior
- Folders per year and month should be created. Within these, the sub-folders should be visible
- All timestamps should be correct

### Actual behavior
{continuation of 5 & 6} 
- The folders were created correctly
- The timestamps are broken 

In detail (example):
test-folder/2014/12 
 → files were created 2 minutes ago… sure 
 → Notable: A file with 1.1MB was created 2 minutes ago, but a file with 946KB was created 7 years ago
 All files below 1MB had correct timestamps; files above had not. (A guess: Are large files split in a certain way, that they lose the timestamp while uploading?)

App/Server(Browser) show the same (incorrect) timestamps^.
Files on the sd-card are untouched, hence still correct.

### Environment data
Android version: 9
Device model: H313
Stock or customized system: stock
Nextcloud app version: 3.16
Nextcloud server version: 21.0.1
Reverse proxy: no

#### Web server error log
No new error-logs around/after testing.
#### Nextcloud log (data/nextcloud.log)
Needed?

edit.: Forgot to upload the screenshot I made from one of the folders
![2021-06-30_22-22-blanked](https://user-images.githubusercontent.com/41995155/124029769-c7f6fe80-d9f5-11eb-885f-dde9e70b0df0.png)

**Comment by AndyScherzinger** [comment]
We do have chunking (upload a file in blocks and letting the server put it together again) while for wifi the chunk size is 10MB and 1MB for cellular. So that seems strange.

@tobiasKaminsky could it be that in case of chunking, the server sets the creation/modification timestamp of the actual time when the file was put together?

**Comment by kralqt** [comment]
Seems related to https://github.com/nextcloud/server/issues/3572
I don't know what library the Android app uses for uploading files, but it needs to set the X-OC-Mtime header for both chunked and non-chunked uploads.

**Comment by AlvaroBrey** [comment]
Fixed by https://github.com/nextcloud/android/pull/9884

**Comment by draakuns** [comment]
Hi @AlvaroBrey. 

This was supposed to be fixed on milestone 3.20.0 by a commit from @tobiasKaminsky, but I am facing it yet on:
Android Nextcloud app 3.20.3
Nextcloud 24.0.3 + MariaDB on Docker ARM64 

I just landed here, but a couple of hours before I posted on NC forums the issue with 2 screenshots (Android and NC datadir)
[NC forums](https://help.nextcloud.com/t/bug-timestamps-not-following-a-rule-it-randomly-changes-file-date-on-files-from-a-dir/142658)

Is it possible to re-check this issue? Can be reopened or shall I open a new one referring this one?

Thanks!
